hooks/_common.sh

```diff
@@ -163,21 +163,30 @@ function common::is_hook_run_on_whole_repo {
 # 2.1. If at least 1 check failed - change exit code to non-zero
 # 3. Complete hook execution and return exit code
 # Arguments:
-#   args (string with array) arguments that configure wrapped tool behavior
 #   hook_id (string) hook ID, see `- id` for details in .pre-commit-hooks.yaml file
+#   args_array_length (integer) Count of arguments in args array.
+#   args (array) arguments that configure wrapped tool behavior
 #   files (array) filenames to check
 #######################################################################
 function common::per_dir_hook {
-  local -r args="$1"
-  local -r hook_id="$2"
+  local -r hook_id="$1"
+  declare -i args_array_length=$2
   shift 2
+  declare -a args=()
+  # Expand args to a true array.
+  # Based on https://stackoverflow.com/a/10953834
+  while ((args_array_length-- > 0)); do
+    args+=("$1")
+    shift
+  done
+
   local -a -r files=("$@")
```

Comment:
For clarity (otherwise we need to be more picky on which positional ARG should be considered a list of files passed in to this function):
```suggestion
  # assign rest of function's positional ARGS into `files` array,
  # despite there's only one positional ARG left
  local -a -r files=("$@")
```